src/app/models/constraint-models/constraint.ts

```diff
@@ -18,12 +20,15 @@ export class Constraint {
   protected _parentConstraint: Constraint;
   // The negation flag indicating whether to add a logical negation to the constraints
   protected _negated: boolean;
+  // The dimension for which observations will be selected
+  private _dimension: string;
```

Comment:
I would expect this to be a field of `CombinationConstraint`. Why is it needed here?